File: src/main/java/fields/FieldSerializer.java
package fields;

import java.util.Set;
import java.util.regex.Pattern;

public class FieldSerializer {
    private static final Pattern INVALID_CHAR_PATTERN = Pattern.compile("[\\[\\] ]");

    public static String serialize(final Set<String> fields) {
        if (fields == null) { return ""; }
        return INVALID_CHAR_PATTERN.matcher(fields.toString()).replaceAll("");
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/fields/FieldSerializer.kt
package com.chrisdempewolf.pinterest.fields

import java.util.regex.Pattern

object FieldSerializer {
    private val INVALID_CHAR_PATTERN = Pattern.compile("[\\[\\] ]")

    @JvmStatic
    fun serialize(fields: Set<String>?): String
        = if (fields == null) ""
          else INVALID_CHAR_PATTERN.matcher(fields.toString()).replaceAll("")

}
